stores/metadata.go

```diff
@@ -1759,6 +1864,12 @@ func (s *SQLStore) UpdateObject(ctx context.Context, bucket, path, contractSet,
 			return fmt.Errorf("UpdateObject: failed to delete object: %w", err)
 		}
 
+		// create the dir
+		dirID, err := makeDirsForPath(tx, path)
+		if err != nil {
+			return fmt.Errorf("failed to create directories: %w", err)
```

Comment:
`nit` would also log the `path` here in the error message or just drop the context because I think `makeDirsForPath` already adds it